### PR TITLE
KAFKA-10679: [Streams] migrate kafka-site updated docs to kafka/docs

### PR DESCRIPTION
**Pull request by bbejeck**
During the AK website upgrade, changes made to kafka-site weren't migrated back to kafka-docs.

This PR is an attempt at porting the streams changes to kafka/docs

For the most part, the bulk of the changes in the PR are cosmetic.

For testing:

1. I reviewed the PR diffs
2. Rendered the changes locally

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
Made another pass over all the docs and found some more fixes.

File: docs/streams/developer-guide/testing.html [hunk 1]
```diff
@@ -73,67 +71,55 @@ <h2><a class="toc-backref" href="#testing-topologytestdriver" title="Permalink t
...
 Topology topology = new Topology();
 topology.addSource("sourceProcessor", "input-topic");
 topology.addProcessor("processor", ..., "sourceProcessor");
-topology.addSink("sinkProcessor", "result-topic", "processor");
+topology.addSink("sinkProcessor", "output-topic", "processor");
 // or
 // using DSL
 StreamsBuilder builder = new StreamsBuilder();
-builder.stream("input-topic").filter(...).to("result-topic");
+builder.stream("input-topic").filter(...).to("output-topic");
```
Comment (vvcephei) [inline docs/streams/developer-guide/testing.html:82]: Hmm, is this supposed to match the output topic in L103 below? That one says "result-topic".

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -135,14 +134,12 @@ <h3><a id="streams_api_changes_260" href="#streams_api_changes_260">Streams API
...
     </p>
-
     <p>
         New end-to-end latency metrics have been added. These task-level metrics will be logged at the INFO level and report the min and max end-to-end latency of a record at the beginning/source node(s)
         and end/terminal node(s) of a task. See <a href="https://cwiki.apache.org/confluence/x/gBkRCQ">KIP-613</a> for more information.
     </p>
-
     <p>
-        As of 2.6.0 Kafka Streams deprecates <code>KStream.through()</code> in favor of the new <code>KStream.repartition()</code> operator
+        As of 2.6.0 Kafka Streams deprecates <code>KStream.through()</code> if favor of the new <code>KStream.repartition()</code> operator
```
Comment (vvcephei) [inline docs/streams/upgrade-guide.html]: Seems like this change is wrong.
```suggestion
        As of 2.6.0 Kafka Streams deprecates <code>KStream.through()</code> in favor of the new <code>KStream.repartition()</code> operator
```

**Review by vvcephei — APPROVED** [review]
Thanks, @bbejeck ! LGTM overall, but I noticed a couple of small errors.

File: docs/streams/developer-guide/testing.html (hunk 1 above)
Comment (bbejeck) [reply docs/streams/developer-guide/testing.html:82]: good catch! Thanks

File: docs/streams/upgrade-guide.html (hunk 2 above)
Comment (bbejeck) [reply docs/streams/upgrade-guide.html]: ack

**Comment by bbejeck** [comment]
Merged #9554 into trunk

**Comment by bbejeck** [comment]
cherry-picked to 2.7

**Pull request merged**